### PR TITLE
GIDS: Enhance the data file upload process. #20052

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
As a GIDS user, I would like to be able to upload files as they are received without having to manipulate them in any way so that extra steps are removed from my process.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/4035

## Acceptance criteria
- [x] 1. Files that are delimited by the following values can be uploaded:
     - comma ,
     - pipe |
- [x] 2. Files that have the following extensions can be uploaded:
     - .csv
     - .txt
- [x] 3. Files that are exported use the default column separator specified for the CVS type in https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/config/csv_file_defaults.yml
- [x] 4. When uploading a file with an opening double quote (") in a column but not a closing double quote (") an error message is displayed letting user know that there is bad data and it is related to missing closing double quote (")

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]

File: app/models/upload.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def csv_type_check?
   end
 
   def check_for_headers
-    return unless upload_file && csv_type && skip_lines
+    return unless upload_file && csv_type && skip_lines && col_sep
```
Comment (kfrz) [inline app/models/upload.rb]: Chaining `&&` like this makes me want to see a dedicated method call, like `return unless condition_is_met`

**Review by kfrz — APPROVED** [review]
This looks good to go, thanks for adding specs! 

I did make one comment re: a stylistic nitpick.

For completeness' sake, @zurbergram would you mind linking issue and updating PR body w/ AC items checked etc.

File: app/models/upload.rb (hunk 1 above)
Comment (zurbergram) [reply app/models/upload.rb]: added method

**Pull request merged**